### PR TITLE
Add Nodeselector to pipelineconfig fix issue #2863

### PR DESCRIPTION
**Pull request by NikeNano**
It is currently not possible to set the nodeselector for a whole pipeline as one operation. Instead the nodeselector have to be applied to each ContainerOP https://github.com/kubeflow/pipelines/blob/c0153a0b199eb4ec1323fc87d5cc31ca92be7dca/sdk/python/kfp/dsl/_container_op.py#L823

This PR intends to fix issue: https://github.com/kubeflow/pipelines/issues/2863 and allow users to set nodeslector for in pipelineconfig. If the user also specifies the nodeslector on the container op level this will be applied by argo: ``` NodeSelector is a selector which will result in all pods of the workflow to be scheduled on the selected node(s). This is able to be overridden by a nodeSelector specified in the template.``` [link](https://github.com/argoproj/argo/blob/f07f7bf61147b3444255117c26bfd38261220e95/pkg/apis/workflow/v1alpha1/workflow_types.go#L168).

### CODE REVIEW

**Comment by Ark-kun** [comment]
/lgtm
Would it be OK is we call the option `default_pod_node_selector` (and `set_default_pod_node_selector`)? This might help disambiguate the workflow-specific options (e.g. TTL), default per-pod options (e.g. ImagePullSecrets) and default per-container options (e.g. resource limits).

**Comment by NikeNano** [comment]
> /lgtm
> Would it be OK is we call the option `default_pod_node_selector` (and `set_default_pod_node_selector`)? This might help disambiguate the workflow-specific options (e.g. TTL), default per-pod options (e.g. ImagePullSecrets) and default per-container options (e.g. resource limits).

Sure, sounds like a good proposal. I will update :)

**Comment by Ark-kun** [comment]
/hold
Travis issue

**Comment by NikeNano** [comment]
@Ark-kun travis seems to be up again, looking at other PR:s. How do I force restart the Travis CI?

File: sdk/python/kfp/compiler/compiler.py [hunk 1]
```diff
@@ -699,6 +704,7 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
         raise ValueError(
                   'Invalid imagePullPolicy. Must be one of `Always`, `Never`, `IfNotPresent`.'
               )
+
```
Comment (NikeNano) [inline sdk/python/kfp/compiler/compiler.py:707]:
```suggestion
```

**Comment by NikeNano** [comment]
@Ark-kun the issues is resolved :)

**Comment by ranajoy-dutta** [comment]
Hi,

I am trying to run a component in a specific nodegroup in EKS. Can anyone guide how can I leverage Node Selector in this script?
Currently I'm able to execute this script in default node group.

    name: Run Analytics
    inputs:
      - {name: Repo Path, type: String}
    metadata:
      annotations:
        author: Ranajoy Dutta <ranajoy@abc.com>
    implementation:
      container:
        image: ubuntu:latest
        command:
          - bash 
          - -exc
          - |
            repo_path=$0
            echo "some code here"
          - {inputValue: Repo Path}

**Pull request merged**